MonoGame.Framework/Android/MonoGameAndroidGameView.cs

```diff
@@ -4,346 +4,919 @@
+                    if (updates > 0) {
+                        var t = updates - (stopWatch.Elapsed.TotalMilliseconds - tick);
+                        if (t > 0) {
+                            if (LogFPS) {
+                                Log.Verbose ("AndroidGameView", "took {0:F2}ms, should take {1:F2}ms, sleeping for {2:F2}", stopWatch.Elapsed.TotalMilliseconds - tick, updates, t);
+                            }
+                            if (token.IsCancellationRequested)
+                                return;
+                        }
+                    }
+                }
+                Log.Verbose ("AndroidGameView", "RenderLoop exited");
+            } catch (Exception ex) {
+                Log.Error ("AndroidGameView", ex.ToString ());
+            } finally {
+                lock (lockObject) {
+                    isExited = true;
+                    cts = null;
+                    if (glSurfaceAvailable)
+                        DestroyGLSurface ();
+                    if (glContextAvailable) {
+                        DestroyGLContext ();
+                        ContextLostInternal ();
+                    }
+                }
+            }
+        }
 
-            _gameWindow.ChangeClientBounds(new Rectangle(0, 0, width, height));
+        DateTime prevUpdateTime;
+        DateTime prevRenderTime;
+        DateTime curUpdateTime;
+        DateTime curRenderTime;
+        FrameEventArgs updateEventArgs = new FrameEventArgs ();
 
-            manager.ApplyChanges();
+        void UpdateFrameInternal (FrameEventArgs e)
+        {
+            OnUpdateFrame (e);
+            if (UpdateFrame != null)
+                UpdateFrame (this, e);
+        }
 
-            SurfaceChanged(holder, format, width, height);
-            Android.Util.Log.Debug("MonoGame", "MonoGameAndroidGameView.SurfaceChanged: format = " + format + ", width = " + width + ", height = " + height);
+        protected virtual void OnUpdateFrame (FrameEventArgs e)
+        {
 
-            if (_game.GraphicsDevice != null)
-                _game.graphicsDeviceManager.ResetClientBounds();
         }
 
-        void ISurfaceHolderCallback.SurfaceDestroyed(ISurfaceHolder holder)
+        // this method is called on the main thread
+        void RunIteration (CancellationToken token)
         {
-            SurfaceDestroyed(holder);
-            Android.Util.Log.Debug("MonoGame", "MonoGameAndroidGameView.SurfaceDestroyed");
+            if (token.IsCancellationRequested)
+                return;
+
+            curUpdateTime = DateTime.Now;
+            if (prevUpdateTime.Ticks != 0) {
+                var t = (curUpdateTime - prevUpdateTime).TotalMilliseconds;
+                updateEventArgs.Time = t < 0 ? 0 : t;
+            }
+            try {
+                UpdateFrameInternal (updateEventArgs);
+            } catch (Content.ContentLoadException) {
+                // ignore it..

```

Comment:
@Jjagg I updated the code to throw the exceptions on the UIThread of the Activity. That in theory should help.